src/services/Credential.js

```diff
@@ -2139,33 +2140,64 @@ class Credential {
 	}
 
 	async ensureDnsValue() {
+		/**
+		 * Define LookupAddress because node doesn't do it
+		 * @typedef dns.LookupAddress
+		 * @property address {String}
+		 * @property family {Number}
+		 */
 
-		const resolveDns = (fqdn) => {
-			const promise = util.promisify(dns.lookup)(fqdn).catch(reason => {
+		/**
+		 * Resolves an fqdn with timeout
+		 * @param fqdn
+		 * @param options
+		 * @returns {Promise<any>}
+		 */
+		function resolveDns(fqdn, options) {
+			const promise = util.promisify(dns.lookup)(fqdn, options).catch(reason => {
 				debug_dns(`Failed to resolve ${fqdn} (${reason})`);
 				throw reason;
 			});
 			return CommonUtils.withTimeout(promise, 3000, new Error('DNS resolution timed out'));
-		};
+		}
+
+		/**
+		 * Validates if the edge can be used
+		 * @param expected_ip {dns.LookupAddress}
+		 * @param real_ip {dns.LookupAddress}
+		 * @param edge_ips { Array.<dns.LookupAddress> | dns.LookupAddress }
+		 */
+		function canUseEdge(expected_ip, real_ip, edge_ips){
```

Comment:
`canUseEdge()` parameter names don't make any sense

Reply:
agree... but local consistency xD that was the name they had, should we change all the names?